core/src/main/java/eu/okaeri/configs/postprocessor/ConfigPostprocessor.java

```diff
@@ -169,6 +178,12 @@ public ConfigPostprocessor updateLinesKeys(@NonNull ConfigSectionWalker walker)
 
             lastIndent = indent;
             String updatedLine = walker.update(line, currentPath.get(currentPath.size() - 1), currentPath);
+
+            while (collectionsStartCount != 0) {
```

Comment:
The logic changes in this class are really convoluted and have a lot of mutation based whiles. This needs better structure, should be very much doable with at least semi-simple result.